### PR TITLE
Add drone, lint

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Fix: #32

### CODE REVIEW

**Comment by tobiasKaminsky** [comment]
@mario this is now ready

File: .drone.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      - sh -c "if [ '${FLAVOR}' != 'Lint' ]; then ./gradlew assemble${FLAVOR}; fi"
+    when:
+      matrix:
+        FLAVOR: [Generic, Gplay]
+
+  lint:
+    image: nextcloudci/android:android-33
+    commands:
+      # needs gplay
+      - sed -i '/.*com.google.*/s/^.*\\/\\///g' build.gradle
```
Comment (mario) [inline .drone.yml]: This is not needed.

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+pipeline:
+  compile:
+    image: nextcloudci/android:android-33
+    commands:
+      # build app and assemble APK
+      - sh -c "if [ '${FLAVOR}' != 'Lint' ]; then ./gradlew assemble${FLAVOR}; fi"
+    when:
+      matrix:
+        FLAVOR: [Generic, Gplay]
```
Comment (mario) [inline .drone.yml]: Is this because Gradle upper-cases first letter? Because the flavours are "generic" and "gplay".

File: .gitignore [hunk 3]
```diff
@@ -55,3 +51,9 @@ freeline.py
 freeline/
 freeline_project_description.json
 .DS_Store
+
+# Android Studio and Gradle specific entries
+.gradle/
+build/
+.idea/
+/gradle.properties
```
Comment (mario) [inline .gitignore]: Can I get a new line pls?

File: scripts/lint/getBranchName.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+# $1: username, $2: password/token, $3: pull request number
+
+if [ -z $3 ] ; then
+    echo "master";
+else
+    curl 2>/dev/null -u $1:$2 https://api.github.com/repos/nextcloud/talk-android/pulls/$3 | grep \"ref\": | grep -v master | cut -d"\"" -f4
+fi
```
Comment (mario) [inline scripts/lint/getBranchName.sh]: New line.

File: .drone.yml (hunk 2 above)
Comment (mario) [reply .drone.yml]: Yup, that's why - ignore me :P

Comment (tobiasKaminsky) [reply .drone.yml]: They are uppercase so I can use them directly in assemble${FLAVOR} (which then becomes assembleGplay / assemble Generic)

**Comment by mario** [comment]
Thanks.

**Pull request merged**